aws/console.go

```diff
@@ -0,0 +1,141 @@
+package aws
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+)
+
+// AWSClient is an interface for AWS operations
+type awsClientInterface interface {
+	GetConsoleURL() (string, error)
+}
+
+// awsClient is the implementation of AWSClient interface
+type awsClient struct {
+	session *session.Session
+}
+
+// NewAWSClient creates a new AWSClient instance
+//
+//	By default NewSession will only load credentials from the shared credentials file (~/.aws/credentials).
+func NewAWSClient() awsClientInterface {
+	// Create session
+	sess := session.Must(session.NewSessionWithOptions(session.Options{
+		SharedConfigState: session.SharedConfigEnable,
+	}))
+
+	return &awsClient{
+		session: sess,
+	}
+}
+
+// GetConsoleURL returns the AWS Management Console URL
+// ref: https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_providers_enable-console-custom-url.html
+func (c *awsClient) GetConsoleURL() (string, error) {
+	amazonDomain := c.getConsoleDomain(*c.session.Config.Region)
+
+	// Create get signin token URL
+	creds, err := c.session.Config.Credentials.Get()
+	if err != nil {
+		return "", errors.New("failed to get AWS session")
+	}
+
+	token, err := c.getSinginToken(creds, amazonDomain)
+	if err != nil {
+		return "", err
+	}
+
+	targetURL := fmt.Sprintf("https://console.%s/console/home", amazonDomain)
+	params := url.Values{
+		"Action":      []string{"login"},
+		"Destination": []string{targetURL},
+		"SigninToken": []string{token},
+	}
+
+	return fmt.Sprintf("https://signin.%s/federation?%s", amazonDomain, params.Encode()), nil
+}
+
+// getConsoleDomain returns the console domain based on the region
+func (c *awsClient) getConsoleDomain(region string) string {
+	var amazonDomain string
+
+	if strings.HasPrefix(region, "us-gov-") {
+		amazonDomain = "amazonaws-us-gov.com"
+	} else if strings.HasPrefix(region, "cn-") {
+		amazonDomain = "amazonaws.cn"
+	} else {
+		amazonDomain = "aws.amazon.com"
+	}
+	return amazonDomain
+}
+
+// getSinginToken retrieves the signin token
+func (c *awsClient) getSinginToken(creds credentials.Value, amazonDomain string) (string, error) {
```

Comment:
`Singing` is typo.

Reply:
fiexd type.
https://github.com/cybozu/assam/pull/258/commits/f908e6b8447af545c8b8029149b965b58d4d75ca